client/src/App.tsx

```diff
@@ -1,20 +1,37 @@
-import { useEffect, useState } from 'react';
+import { useState } from 'react';
 import React from 'react';
-import { Routes, Route } from 'react-router-dom';
-import logo from './logo.svg';
+import { Routes, Route, useNavigate } from 'react-router-dom';
 import './App.css';
 import NavBar from './components/NavBar';
 import TripEntryForm from './Pages/TripEntryForm';
 import SavedTrips from './Pages/SavedTrips';
+import TripDetails from './Pages/TripDetails';
+import TripEditForm from './Pages/TripEditForm';
 import NotFound from './Pages/NotFound';
+import { placeholder, TripEntry } from './lib/data';
 
 function App() {
+  const [editTrip, setEditTrip] = useState(placeholder);
+  const navigate = useNavigate();
+
+  function handleEditTripSubmit(trip: TripEntry) {
+    setEditTrip(trip);
```

Comment:
I don't think you need to keep the `editTrip` in state since you will get it from the route params.

But, this isn't a bad solution. It has its advantages. To do it only with route params, you would remove the editTrip state and just navigate in response to TripDetails onClick. Then in TripEditForm, instead of passing the editTrip as a prop, you would call `const { tripId } = useParams();` to get the trip ID.